File: src/main/java/com/ascargon/rocketshow/midi/MidiControl.java
package com.ascargon.rocketshow.midi;

import com.ascargon.rocketshow.util.ControlAction;
import jakarta.xml.bind.annotation.XmlRootElement;

/**
 * Map one specific MIDI event to an action.
 *
 * @author Moritz Vieli
 */
@XmlRootElement
public class MidiControl extends ControlAction {

	// If null -> all channels
	private Integer channelFrom;

	// The note, this action should be triggered. If null -> all notes
	private Integer noteFrom;

	public Integer getChannelFrom() {
		return channelFrom;
	}

	public void setChannelFrom(Integer channelFrom) {
		this.channelFrom = channelFrom;
	}

	public Integer getNoteFrom() {
		return noteFrom;
	}

	public void setNoteFrom(Integer noteFrom) {
		this.noteFrom = noteFrom;
	}

}

File: src/main/webapp/src/app/models/midi-control.ts
import { ControlAction } from './control-action';

export class MidiControl extends ControlAction {
    channelFrom: number = 0;
    noteFrom: number = 0;

    constructor(data?: any) {
        super(data);

        if(!data) {
        	return;
        }

        this.channelFrom = data.channelFrom;
        this.noteFrom = data.noteFrom;
    }
}
